### PR TITLE
Remove temporary directory after using it (if it's empty)

### PR DESCRIPTION
**Pull request by AlexVanderbist**
See #11

### CODE REVIEW

**Comment by danhunsaker** [comment]
It occurs to me we should probably be using Flysystem calls instead of directly interacting with the local filesystem.  For one, Glide uses Flysystem exclusively for things like this, so consistency would be helpful.  But for two, should the user swap out the storage for something other than Local, everything relying on direct filesystem access would probably break, and at least not work correctly.

Probably affects more than just the new code in this PR, though, so it might need a separate PR...

File: src/GlideConversion.php [hunk 1]
```diff
@@ -95,7 +96,15 @@ public function save(string $outputFile)
             return;
         }
 
+        $conversionResultDirectory = pathinfo($this->conversionResult, PATHINFO_DIRNAME);
+
         rename($this->conversionResult, $outputFile);
+
+        $iterator = new FilesystemIterator($conversionResultDirectory);
+
+        if (! $iterator->valid()) {
```
Comment (freekmurze) [inline src/GlideConversion.php]: Could this be replace with `is_dir`?

**Comment by freekmurze** [comment]
@danhunsaker I think using the common file system functions is fine in this case. The chance is fairly low that somebody doesn't want to store the temp files locally.

The philosophy behind this package is that from the outside it should matter what we use internally to convert the images. Note that the `GlideFactory` itself is marked as `final` and almost every class var and method is `private`. In this way we're hinting that you should bother modifying this class.

**Comment by danhunsaker** [comment]
To that I can only comment that I'd (still) personally like to at least have the option of using the Memory storage instead of local disk.  That said, absolutely not my call.  :smile:

File: src/GlideConversion.php (hunk 1 above)
Comment (AlexVanderbist) [inline src/GlideConversion.php]: The `FilesystemIterator` is used here to check if there are any files left in the directory, not to check whether it's a directory

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**